ws_scaffold_raft test failure

https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=9031&view=logs&j=5435e0ac-25e5-5426-50be-61b0d0ea8d34&t=63d660c3-6f76-573a-0154-fd0bcb096ded

55: 2020-06-08 06:24:46.873 | INFO     | infra.remote:stop:448 - [127.243.246.33] closing
55: 2020-06-08 06:24:46.937 | INFO     | infra.ccf:stop_all_nodes:384 - All nodes stopped...
55: Traceback (most recent call last):
55:   File "/mnt/build/1/s/tests/ws_scaffold.py", line 63, in <module>
55:     run(args)
55:   File "/mnt/build/1/s/tests/ws_scaffold.py", line 49, in run
55:     test(network, args, notifications_queue)
55:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 19, in wrapper
55:     return func(*args, **kwargs)
55:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 42, in wrapper
55:     return func(network, args, *nargs, **kwargs)
55:   File "/mnt/build/1/s/tests/suite/test_requirements.py", line 42, in wrapper
55:     return func(network, args, *nargs, **kwargs)
55:   File "/mnt/build/1/s/tests/ws_scaffold.py", line 30, in test
55:     assert r.result == True, r.result
55: AssertionError: None
38/57 Test #55: ws_scaffold_raft .................***Failed    3.63 sec

```
2020-06-26 12:13:23.330 | DEBUG    | infra.clients:log_response:166 - 404 @0.0 Session 'app' is not open.

2020-06-26 12:13:23.330 | INFO     | infra.remote:stop:479 - [127.148.83.86] closing
2020-06-26 12:13:23.394 | INFO     | infra.remote:stop:479 - [127.1.63.155] closing
2020-06-26 12:13:23.458 | INFO     | infra.ccf:stop_all_nodes:388 - All nodes stopped...
Traceback (most recent call last):
  File "/mnt/build/2/s/tests/ws_scaffold.py", line 63, in <module>
    run(args)
  File "/mnt/build/2/s/tests/ws_scaffold.py", line 49, in run
    test(network, args, notifications_queue)
  File "/mnt/build/2/s/tests/suite/test_requirements.py", line 19, in wrapper
    return func(*args, **kwargs)
  File "/mnt/build/2/s/tests/suite/test_requirements.py", line 42, in wrapper
    return func(network, args, *nargs, **kwargs)
  File "/mnt/build/2/s/tests/suite/test_requirements.py", line 42, in wrapper
    return func(network, args, *nargs, **kwargs)
  File "/mnt/build/2/s/tests/ws_scaffold.py", line 30, in test
    assert r.result == True, r.result
AssertionError: None
```

--- a/src/node/rpc/node_frontend.h
+++ b/src/node/rpc/node_frontend.h
@@ -316,6 +316,18 @@ namespace ccf
         "quotes", HTTP_GET, json_read_only_adapter(get_quotes))
         .set_auto_schema<GetQuotes>()
         .install();
+      auto network_status = [this](auto& args, nlohmann::json&&) {
+        auto service_view = args.tx.get_read_only_view(network.service);
+        auto service_state = service_view->get(0);
+        if (service_state.has_value())
+        {
+          return make_success(service_state.value().status);
+        }
+        return make_error(HTTP_STATUS_NOT_FOUND, "Network status is unknown");
+      };
+      make_read_only_endpoint(
+        "network", HTTP_GET, json_read_only_adapter(network_status))
+        .install();
     }
   };
 

--- a/tests/infra/clients.py
+++ b/tests/infra/clients.py
@@ -481,7 +481,7 @@ def client(
     ca=None,
     description=None,
     log_file=None,
-    prefix="users",
+    prefix="app",
     binary_dir=".",
     connection_timeout=3,
     request_timeout=3,

--- a/tests/ws_scaffold.py
+++ b/tests/ws_scaffold.py
@@ -6,6 +6,7 @@ import infra.notification
 import infra.net
 import suite.test_requirements as reqs
 import infra.e2e_args
+import time
 
 from loguru import logger as LOG
 
@@ -22,6 +23,19 @@ def test(network, args, notifications_queue=None):
         for i in [1, 50, 500]:
             r = c.rpc("log/private", {"id": 42, "msg": msg * i})
             assert r.result == True, r.result
+
+    # Before we start sending transactions to the secondary,
+    # we want to wait for its app frontend to be open, which is
+    # when it's aware that the network is open.
+    end_time = time.time() + 10
+    with other.node_client() as nc:
+        while time.time() < end_time:
+            r = nc.get("network")
+            if r.result == "OPEN":
+                break
+            else:
+                time.sleep(0.1)
+        assert r.result == "OPEN", r
 
     LOG.info("Write on secondary through forwarding")
     with other.user_client(ws=True) as c:

